command/agent/template/template_test.go

```diff
@@ -400,6 +405,70 @@ func TestServerRun(t *testing.T) {
 	}
 }
 
+// TestNewServerLogLevels tests that the server can be started with any log
+// level.
+func TestNewServerLogLevels(t *testing.T) {
+	ts := createHttpTestServer()
+	defer ts.Close()
+
+	tmpDir, err := ioutil.TempDir("", "agent-tests")
```

Comment:
Since we're using Go 1.17+, it might be a good opportunity to swap this call to `os. MkdirTemp` in here and elsewhere in this file since that's what it really calls [under the hood](https://pkg.go.dev/io/ioutil#TempDir).

The doc mentions that it is preferred to call funcs from `os` over `ioutil`:
> As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details.